internal/controller/git/work_tree.go

```diff
@@ -419,6 +424,118 @@ func (w *workTree) ListCommits(limit, skip uint) ([]CommitMetadata, error) {
+	CreationDate time.Time
+	// Author is the author of the latest commit on the branch, in the format "Name <email>".
+	Author string
+	// Committer is the person who committed the latest commit on the branch, in the format "Name <email>".
+	Committer string
+	// Subject is the subject (first line) of the latest commit message on the branch.
+	Subject string
+}
+
+// TODO: Remove this debug function
+func (w *workTree) debugListBranches(cmd *exec.Cmd, references string) {
+	fmt.Println("\n=====\n[BEBUG] [debugListBranches] url: ", w.url)
+
+	if cmd != nil {
+		fmt.Println("[BEBUG] [debugListBranches] cmd: ", cmd.String())
+		res, _ := libExec.Exec(cmd)
+		fmt.Println("[BEBUG] [debugListBranches] output:\n", string(res))
+	}
+
+	if references != "" {
+		fmt.Println("[DEBUG] [debugListBranches] references: ", references)
+		// Get the list of branches sorted by creation date and only fetch branches (not tags)
+		branchesBytes, _ := libExec.Exec(w.buildGitCommand(
+			"for-each-ref",
+			"--sort=creatordate",
+			"--format=`%(refname:short)|*|%(objectname)|*|%(contents:subject)|*|%(authorname) %(authoremail)|*|%(committername) %(committeremail)|*|%(creatordate:iso8601)`",
+			references,
+		))
+
+		prettyJSON, _ := json.MarshalIndent(string(branchesBytes), "", "  ")
+		fmt.Println("[DEBUG] [debugListBranches] branches:\n", string(prettyJSON))
+	}
+}
+
+func (w *workTree) ListBranches() ([]BranchMetadata, error) {
+	// TODO: remove these debug calls
+	w.debugListBranches(w.buildGitCommand("fetch", "origin"), "refs/remotes")
+	w.debugListBranches(w.buildGitCommand("fetch", "origin", "'refs/heads/feature/*:refs/remotes/origin/feature/*'"), "refs/remotes")
+	w.debugListBranches(w.buildGitCommand("ls-remote", "--heads", "origin"), "")
+	w.debugListBranches(w.buildGitCommand("branch", "-a"), "")
+	w.debugListBranches(nil, "refs/tags")
+	w.debugListBranches(w.buildGitCommand("fetch", "origin", "--tags"), "refs/tags")
+
+	// Fetch the latest references for the branches.
+	if _, err := libExec.Exec(w.buildGitCommand("fetch", "origin")); err != nil {
+		return nil, fmt.Errorf("error fetching branches from repo %q: %w", w.url, err)
+	}
+
+	// This format will output the following fields, separated by `|*|`:
+	// - branch name
+	// - commit ID (latest commit on the branch)
+	// - subject of the latest commit
+	// - author name and email
+	// - committer name and email
+	// - creation date (using `creatordate` for the branch creation)
+	const branchFormat = `%(refname:short)|*|%(objectname)|*|%(contents:subject)|*|%(authorname) %(authoremail)|*|%(committername) %(committeremail)|*|%(creatordate:iso8601)`
+
+	// TODO: Remove this debug print
+	fmt.Println("[BEBUG] [ListBranches] starting ...")
+
+	// Get the list of branches sorted by creation date and only fetch branches (not tags)
+	branchesBytes, err := libExec.Exec(w.buildGitCommand(
+		"for-each-ref",
+		"--sort=creatordate",
+		"--format="+branchFormat,
+		"refs/remotes",
+	))
```

Comment:
@krancour
Sorry for tagging you on that.

When I run from docker desktop
`refs/tags` works
```
git fetch origin --tags
git for-each-ref --format='%(refname)' refs/tags
```

but `refs/remotes`
```
git fetch origin
git for-each-ref --format='%(refname)' refs/remotes
```

I tried both commands on my Mac, and they worked fine. Hopefully, you have some insight into why `refs/remotes` doesn't work in side Kargo pod.

Thanks in advance!